src/rdb/rdb_raft.c

```diff
@@ -1380,18 +1382,33 @@ rdb_raft_cb_notify_membership_event(raft_server_t *raft, void *udata, raft_node_
 }
 
 static void
-rdb_raft_cb_debug(raft_server_t *raft, raft_node_t *node, void *arg,
-		  const char *buf)
+rdb_raft_cb_log(raft_server_t *raft, raft_node_t *node, void *arg, raft_loglevel_e level,
+		const char *buf)
 {
 	struct rdb *db = raft_get_udata(raft);
+	d_rank_t    rank;
 
-	if (node != NULL) {
-		struct rdb_raft_node *rdb_node = raft_node_get_udata(node);
+	if (node == NULL)
+		rank = CRT_NO_RANK;
+	else
+		rank = ((struct rdb_raft_node *)raft_node_get_udata(node))->dn_rank;
 
-		D_DEBUG(DB_TRACE, DF_DB": %s: rank=%u\n", DP_DB(db), buf,
-			rdb_node->dn_rank);
-	} else {
-		D_DEBUG(DB_TRACE, DF_DB": %s\n", DP_DB(db), buf);
+	switch (level) {
+	case RAFT_LOG_ERROR:
+		D_ERROR(DF_DB ": %s: rank=%u\n", DP_DB(db), buf, rank);
+		break;
+	case RAFT_LOG_INFO:
+		/*
+		 * Demote to D_DEBUG, because we don't have a mechanism yet to
+		 * eventually stop replicas that have been excluded from the
+		 * pool. Every 1--2 election timeouts, these replicas will log a
+		 * few election messages, which might attract complaints if done
+		 * with D_INFO.
+		 */
+		D_DEBUG(DB_MD, DF_DB ": %s: rank=%u\n", DP_DB(db), buf, rank);
+		break;
+	default:
+		D_DEBUG(DB_IO, DF_DB ": %s: rank=%u\n", DP_DB(db), buf, rank);
```

Comment:
is the change from DB_TRACE to DB_IO because it is more likely for an admin to configure daos_engine with DD_MASK=io (than DD_MASK=trace)?

Reply:
@kccain, yes. Last week, I tried `dmg server set-logmasks -l <nodes> -m INFO,rdb=DEBUG -d md,io` online to simulate what we might request users to do, and it seemed to work, though not as clear & simple as I wish it to be.